src/Resources/app/administration/src/module/swag-migration/service/swag-migration-error-resolution.service.ts

```diff
@@ -0,0 +1,712 @@
+
+        if (!entityField || entityField.type !== DATA_TYPES.ASSOCIATION) {
+            return false;
+        }
+
+        const relationType = entityField.relation;
+
+        return relationType === HANDLED_RELATION_TYPES.ONE_TO_MANY || relationType === HANDLED_RELATION_TYPES.MANY_TO_MANY;
+    }
+
+    /**
+     * gets the effective entity field to use for a field.
+     * for id fields with associations, returns the association field instead.
+     */
+    getEffectiveEntityField(entityName: string | null | undefined, fieldName: string | null | undefined): Property | null {
+        const correspondingAssociation = this.findCorrespondingAssociationField(entityName, fieldName);
+
+        if (correspondingAssociation) {
+            return correspondingAssociation;
+        }
+
+        return this.getEntityField(entityName, fieldName);
+    }
+
+    /**
+     * determines the field type for rendering (either component type or relation type).
+     */
+    getFieldType(entityName: string | null | undefined, fieldName: string | null | undefined): string | null {
+        const entityField = this.getEntityField(entityName, fieldName);
+
+        if (!entityField || UNHANDLED_FIELD_TYPES.includes(entityField.type as (typeof UNHANDLED_FIELD_TYPES)[number])) {
+            return null;
+        }
+
+        // return relation type for association fields
+        if (entityField.type === DATA_TYPES.ASSOCIATION && entityField.relation) {
+            return entityField.relation as string;
+        }
+
+        // return relation type for uuid fields with corresponding association fields
+        if (entityField.type === DATA_TYPES.UUID) {
+            const correspondingAssociation = this.findCorrespondingAssociationField(entityName, fieldName);
+
+            if (correspondingAssociation?.relation) {
+                return correspondingAssociation.relation as string;
+            }
+        }
+
+        return FIELD_TYPE_COMPONENT_MAPPING[entityField.type as keyof typeof FIELD_TYPE_COMPONENT_MAPPING] ?? null;
+    }
+
+    /**
+     * sorts fields based on predefined priority. Fields with higher priority appear first.
+     * returns a new sorted array without mutating the input.
+     */
+    sortFieldsByPriority(fields: string[]): string[] {
+        return [...fields].sort((a, b) => {
+            const priorityA = PRIORITY_FIELD_MAP.get(a);
+            const priorityB = PRIORITY_FIELD_MAP.get(b);
+
+            // both fields have defined priorities
+            if (priorityA !== undefined && priorityB !== undefined) {
+                return priorityA - priorityB;
+            }
+
+            // only field A has a defined priority
+            if (priorityA !== undefined) {
+                return -1;
+            }
+
+            // only field B has a defined priority
+            if (priorityB !== undefined) {
+                return 1;
+            }
+
+            return 0;
```

Comment:
you could still sort by field names here instead of leaving their order unspecified if no priority was found for both of them

```suggestion
           if (a < b) return -1;
           if (a > b) return 1;
           return 0;
```